doc/system_requirements.md

```diff
@@ -24,20 +24,32 @@ algorithms such as machine learning methods.
 Features are the highest level requirements in this document that describe the 
 main functionality of Advanced Analytics Framework.
 
-### Event Loop
-`feat~event-loop~1`
+### Global Loop
+`feat~global-loop~1`
 
-Event Loop processes only the state transitions by executing queries returned by 
-the Event Handler. Some complex algorithms need a global loop. In each iteration 
-of the loop they need to execute SQL queries until the algorithm no longer 
-needs to run further queries. 
+Some complex algorithms need a global loop. In each iteration of the loop, they 
+need to execute SQL queries until the algorithm no longer needs to run 
+further queries. 
 
 Needs: req
 
-### Event Handler
-`feat~event-handler~1`
+### Implementation Framework
+`feat~implementation-framework~1`
 
-TODO
+The logic of the algorithm is implemented by user. The implementation needs 
+to call a framework which provides certain features to the user code, such as 
+automatic state handling. 
```

Comment:
Here it is better to use things like temporary table handling and running the queries and returning the results instead of the automatic state handling, because the state handling is only necessary in our design. While temporary table handling and running the queries and returning the results, are the things the framework should take care of for the user and which we potentially want replace with different implementations.